### PR TITLE
feat(unify): launchpad header breadcrumbs and reusable tooltip component

### PR DESCRIPTION
**Pull request by mapsandapps**
- Closes [UNIFY-1257]

### User facing changelog

* have a new, last breadcrumb for e2e vs ct
* project name should link back to testing type selection screen
* display branch
* branch is truncated if needed, and has a tooltip
* made a reusable tooltip component and replaced our old ones with it in several places

Screen Shot 2022-04-12 at 10 15 32 AM
Screen Shot 2022-04-12 at 10 15 46 AM

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

[UNIFY-1257]: https://cypress-io.atlassian.net/browse/UNIFY-1257?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by marktnoonan — CHANGES_REQUESTED** [review]
Looks good overall, a couple of changes requested around accessibility and guarding against clearing the testing type while config is loading.

It's also a good time to confirm with extra dynamic content from users in the header now, how comfortable are we with this wrapping that we see here? That might not need to be addressed in this PR, though, until we have a general direction about wrapping vs truncating.

Screen Shot 2022-03-18 at 7 34 48 AM

File: packages/frontend-shared/src/gql-components/HeaderBarContent.vue [hunk 1]
```diff
@@ -28,7 +31,28 @@
           v-if="props.gql?.currentProject"
           class="h-16px mr-2px min-w-16px icon-dark-gray-200"
         />
-        <span class="text-body-gray-700">{{ props.gql?.currentProject?.title }}</span>
+        <span
```
[the remark this answers is not in the document]
Comment (mapsandapps) [reply packages/frontend-shared/src/gql-components/HeaderBarContent.vue]: I didn't think you should ever have an `a` without an `href`, so this was news to me!

In my update, I went ahead and used <nav> and <ol> for this, which i had been intending to do if i made a reusable breadcrumb component. LMK what you think about this.

[the remark this answers is not in the document]
Comment (marktnoonan) [reply packages/frontend-shared/src/gql-components/HeaderBarContent.vue]: Nice, I think that's good. I had only found the disabled link pattern recently, I think we needed it somewhere else in launchpad as well.

**Comment by emilyrohrbough** [comment]
@mapsandapps Any update on this?

**Comment by mapsandapps** [comment]
@emilyrohrbough oops, good call! i've updated the description and the title as well

**Comment by marktnoonan** [comment]
I had some trouble getting the tooltip to dismiss: https://www.loom.com/share/7ee0e62619ce45f98ae4e5a47b7ee6cc

**Comment by marktnoonan** [comment]
Thanks for clarifying @mapsandapps, sorry to be slow responding here. You're right about the current implementation of the tooltips, 9.x was different and while I think we want that experience in 10.0, it's not at all in scope for this PR, so I'll make a new ticket to track it.

Going to pull this down and review properly now.

https://user-images.githubusercontent.com/8340719/163629821-f96fb2e9-c55f-43aa-b30a-544bdd9e6f5e.mov

**Review by marktnoonan — COMMENTED** [review]
These changes work, and I'm delighted we can copy text from these tooltips! I think we should merge this PR as is (once conflicts are resolved), though I see some accessibility issues to address when we are back in the tooltips (which is really soon, and making a perfect tooltip for all cases was not the goal of this PR). The element you hover over should show the tooltip when you focus with the keyboard. But only in "focus-visible" type situations, not if focus lands on an element by clicking.

One other minor thing I noticed the "link" to the project breadcrumb can be clicked even when the browser is already open. It works fine, just seems a little less clear than the "switch testing type" button that does the same thing. I feel like there's just a little tiny window where that breadcrumb being a link is functional, but that's a separate topic.

I'll hit that ✔️  when the conflicts are resolved and this is ready to merge.

**Review by lmiller1990 — APPROVED** [review]

**Comment by lmiller1990** [comment]
Nice, looks great 👍 

Should we hit approve on Percy?

**Review by marktnoonan — APPROVED** [review]

**Pull request merged**